### PR TITLE
Resolve links to workspace files in terminal

### LINKED ISSUE #13047: Path that resolve to files in the workspace are not navigatable
### Feature Description:
In VSCode you can use paths printed in the console (e.g when transpile fails) to navigate to these files. This is a very nice feature.

### PR DESCRIPTION
**Pull request by AlexandraBuzila**
#### What it does
Try and provide links for paths pointing to workspace files in the terminal. The links are created by searching for matching files based on the provided path.

Fixes #13047

Contributed on behalf of STMicroelectronics

#### How to test
Open the theia workspace and echo some partial paths in the terminal. All paths that can be resolved to a workspace file should have valid links that can be used to open the files from the terminal, e.g.:
* ../../packages/terminal/src/browser/terminal-file-link-provider.ts:17:20
* src/browser/terminal-file-link-provider.ts

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by planger — COMMENTED** [review]
Thank you @AlexandraBuzila! This definitely improves the file linking in the terminal and the code looks fine to me.

The only thing I miss is that simple file names don't work (e.g. `package.json`). They only work if they start with `./` as in `./package.json`. Is that something we could fix easily? This seems to work in VS Code.

Probably beyond the scope of this PR, but VS Code is a bit more clever when choosing the CWD. VS Code seems to remember the CWD of each command output in the terminal history and keeps it, whereas we switch for the entire terminal. This is nice of course, if you ran a build and then cd'd into another folder, it keeps the correct linking. If this is easy to do (which I don't expect), it may be worth adding. Otherwise, using the current CWD should be good enough for now.

I have added to OPTIONAL remarks inline. If you disagree, feel free to ignore them.

**Comment by AlexandraBuzila** [comment]
Thank you @planger for the review!
I pushed a new commit with added support for links to paths that don't have any separators. Could you recheck the code, please?

I also had a look at improving the support for CWD and it wasn't so straightforward. It might be worth investigating this in a separate ticket.

File: packages/terminal/src/browser/terminal-file-link-provider.ts [hunk 1]
```diff
@@ -97,8 +129,11 @@ export class FileLinkProvider implements TerminalLinkProvider {
         if (!toOpen) {
             return;
         }
-
         const position = await this.extractPosition(match);
+        return this.openURI(toOpen, position);
+    }
+
+    async openURI(toOpen: URI, position: Position): Promise<void> {
```
[the remark this answers is not in the document]
Comment (AlexandraBuzila) [reply packages/terminal/src/browser/terminal-file-link-provider.ts:149]: The openURI is a new method. Where you referring to it as breaking the API or was there something else that I maybe overlooked?

[the remark this answers is not in the document]
Comment (planger) [reply packages/terminal/src/browser/terminal-file-link-provider.ts:149]: Oh, sorry... my bad. I don't know why I considered this as breaking the API originally. This is completely fine.

**Review by planger — APPROVED** [review]
Thank you for the update @AlexandraBuzila! Works great and code looks good. Thank you!

**Pull request merged**